src/param.rs

```diff
@@ -970,28 +988,24 @@ impl AudioParamProcessor {
         let clamped = self.intrisic_value.clamp(self.min_value, self.max_value);
         self.current_value.store(clamped);
 
-        // no even in timeline and buffer is clean, we can just reuse current buffer,
-        // instead of clearing to just refill with same values
-        if self.event_timeline.is_empty() && self.is_buffer_clean {
-            return;
-        }
-
-        // Define if intrisic buffer can be considered be clean in next call.
-        // If the timeline is empty, we know the buffer will be filled with a
-        // constant value. @note: there could other valid cases
-        self.is_buffer_clean = self.event_timeline.is_empty();
-
-        // populate the buffer for this block
+        // clear the buffer for this block
         self.buffer.clear();
 
-        let next_block_time = dt.mul_add(count as f64, block_time);
         let is_a_rate = self.is_a_rate.load(Ordering::SeqCst);
         let is_k_rate = !is_a_rate;
+        let is_timeline_empty = self.event_timeline.is_empty();
 
-        if is_k_rate {
-            self.buffer.resize(count, self.intrisic_value);
+        if is_k_rate || is_timeline_empty {
+            self.buffer.resize(1, self.intrisic_value);
+
+            // nothing to compute in timeline
+            if is_timeline_empty {
+                return;
+            }
         }
 
+        let next_block_time = dt.mul_add(count as f64, block_time);
+
         loop {
             let some_event = self.event_timeline.peek();
 
```

Comment:
If the next event is at `> next_block_time` and we are not currently in a `setTargetAtTime`, we can also return early with a single valued buffer, no?

Reply:
Hum, actually that's a bit more tricky, because for example for `LinearRamp` or `ExponentialRamp` the actual event time is the end time, their `start_time` being `last_event.time` so this check alone would break them.

I think we could maybe do better, but I need to write all the possible cases down, they don't fit in my head :)